File: commons-aws/src/main/java/pl/wrzasq/commons/aws/sqs/SimpleEventHandler.java
/*
 * This file is part of the pl.wrzasq.commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2019 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package pl.wrzasq.commons.aws.sqs;

import java.util.function.Consumer;

/**
 * SQS event handler that simply processes plain message body.
 */
public class SimpleEventHandler extends EventHandler {
    /**
     * Initializes notification handler.
     *
     * @param messageBodyHandler SQS message handler
     */
    public SimpleEventHandler(Consumer<String> messageBodyHandler) {
        super(data -> messageBodyHandler.accept(data.getBody()));
    }
}

File: commons-aws/src/main/kotlin/pl/wrzasq/commons/aws/messaging/sqs/SimpleEventHandler.kt
/*
 * This file is part of the pl.wrzasq.commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2019, 2021 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package pl.wrzasq.commons.aws.messaging.sqs

/**
 * SQS event handler that simply processes plain message body.
 *
 * @param messageBodyHandler SQS message handler
 */
open class SimpleEventHandler(
    messageBodyHandler: (String) -> Unit
) : EventHandler({ messageBodyHandler(it.body) })
